app/app_test.go

```diff
@@ -320,68 +319,77 @@ func assertServerHealth(t *testing.T, ctx context.Context, opts client.Options)
 	}
 }
 
-func TestCreateDataDirectory(t *testing.T) {
+func TestCreateDataDirectory_MissingDirectory(t *testing.T) {
+	temporalCLI := app.BuildApp("")
+	// Don't call os.Exit
+	temporalCLI.ExitErrHandler = func(_ *cli.Context, _ error) {}
+
+	portProvider := sconfig.NewPortProvider()
+	var (
+		port = portProvider.MustGetFreePort()
+	)
```

Comment:
```suggestion
	port := portProvider.MustGetFreePort()
```

(not really reviewing the PR, just saw this as I passed by)